### ISSUE TITLE
Monitoring and controlling an Itho Daalderop CVE RFT

### ISSUE THREAD

**Issue by reneklootwijk** [Evsdd/The-Evohome-Protocol#5]
By coincidence I noticed my Itho Daalderop CVE RFT uses the RAMSES-II protocol for communication with the RF remotes.

The remote has 4 keys and the 4th key activates a timer. The timer depends on the number of key presses (1-3). When a button is pressed it broadcast a message 3 times.

For mode 1 (low) the following messages are send:

```text
067  I 132 --:------ --:------ 63:163409 22F1 003 000204
```
63:163409 is the address of the remote, my other remote uses: 62:228945

For mode 2 (normal) the following message is send:

```text
060  I 133 --:------ --:------ 63:163409 22F1 003 000304
```

For mode 3 (high) the following message is send:

```text
065  I 134 --:------ --:------ 63:163409 22F1 003 000404
```

For mode 4 (timed high) the following messages are send for 1 key press (timer=10m):

```text
067  I 135 --:------ --:------ 63:163409 22F1 003 000204
053  I 136 --:------ --:------ 63:163409 22F3 003 00000A
```

For mode 4 (timed high) the following messages are send for 2 key presses (timer=20m):

```text
067  I 137 --:------ --:------ 63:163409 22F1 003 00000A
053  I 138 --:------ --:------ 63:163409 22F3 003 000014
```

For mode 4 (timed high) the following messages are send for 3 key presses (timer=30m):

```text
067  I 137 --:------ --:------ 63:163409 22F1 003 000014
053  I 138 --:------ --:------ 63:163409 22F3 003 00001E
```

Controlling also works by sending the following for instance:

```text
I --- 18:000730 --:------ 63:163409 22F1 003 000404
```

**Comment by zxdavb** [comment]
Please format your post - I have done a start for you.

**Comment by zxdavb** [comment]
Have a look at:
 - `22F1`: https://github.com/zxdavb/evohome_rf/blob/97e4b3f28d67d93050aca9e147c2c7f849c9d637/evohome_rf/parsers.py#L1299-L1323
  - `22F3`: https://github.com/zxdavb/evohome_rf/blob/97e4b3f28d67d93050aca9e147c2c7f849c9d637/evohome_rf/parsers.py#L1326-L1334

Also look at `31D9`, and `31DA`.

You can capture packet logs and try things like:
```
cat packet.log | python client.py parse
````

You can post to @Raqbit here: https://gitter.im/evohome_rf/community

He & you may be able to collaborate further.

I have switched to an MIT license, here and at **evohome_rf** - enjoy!

**Comment by reneklootwijk** [comment]
Apparantly you already knew part of it. I will now try to pair my code as a new remote to the Itho CVE.

My node.js module is alsmost finished, so I use my own code to monitor and try things.

I see 31D9 and 31DA messages from toher unknown devices, with id 20: and 37:. Does this sound familiar.

BTW thanks for using the MIT license!

**Comment by zxdavb** [comment]
> I see 31D9 and 31DA messages from toher unknown devices, with id 20: and 37:. Does this sound familiar.

Yes - have a look in **evohome_rf/parsers.py** - all that is known by me is in there!

(except for the sequence # issue)

@Raqbit may know more
